sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5969,36 +6165,7 @@ pi_result piEnqueueEventsWaitWithBarrier(pi_queue Queue,
 
   // Lock automatically releases when this goes out of scope.
   std::scoped_lock lock(Queue->Mutex);
-
-  bool UseCopyEngine = false;
-
-  _pi_ze_event_list_t TmpWaitList;
-  if (auto Res = TmpWaitList.createAndRetainPiZeEventList(
-          NumEventsInWaitList, EventWaitList, Queue, UseCopyEngine))
-    return Res;
-
-  // Get a new command list to be used on this call
-  bool OkToBatch = true;
-  pi_command_list_ptr_t CommandList{};
-  if (auto Res = Queue->Context->getAvailableCommandList(
-          Queue, CommandList, UseCopyEngine, OkToBatch))
-    return Res;
-
-  ze_event_handle_t ZeEvent = nullptr;
-  auto Res = createEventAndAssociateQueue(Queue, Event, PI_COMMAND_TYPE_USER,
-                                          CommandList);
-  if (Res != PI_SUCCESS)
-    return Res;
-  ZeEvent = (*Event)->ZeEvent;
-  (*Event)->WaitList = TmpWaitList;
-
-  ZE_CALL(zeCommandListAppendBarrier,
-          (CommandList->first, ZeEvent, (*Event)->WaitList.Length,
-           (*Event)->WaitList.ZeEventList));
-
-  // Execute command list asynchronously as the event will be used
-  // to track down its completion.
-  return Queue->executeCommandList(CommandList, false, OkToBatch);
+  return Queue->insertBarrier(NumEventsInWaitList, EventWaitList, Event);
```

Comment:
This is the only call to this function, is there any good reason for it not to be just inlined here? A side-effect of this would be that there would be no confusions between `insertActiveBarriers` and `insertBarrier`

Reply:
Since it is an operation on the queue as a whole, I figured having it a member function seemed fitting and would also make it easier for other functions that might need implicit barriers to use it, though I have no concrete examples of where that could apply. I will inline it ASAP.

Reply:
`insertBarrier` has been removed and the body has been inlined into `piEnqueueEventsWaitWithBarrier`.

Reply:
Thanks!